src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -207,6 +208,14 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return this._onClientSessionReady.event;
 	}
 
+	public getApplicableConnectionProviderIds(kernelDisplayName: string): string[] {
+		let ids = [];
+		if (kernelDisplayName) {
+			ids = this._kernelDisplayNameToConnectionProviderIds.get(kernelDisplayName);
+		}
+		return !ids ? [] : ids;
```

Comment:
Can't we return 'ids' directly without a condition since it was initialized already? Do we get undefined value ?

Reply:
I'd rather return an empty array than return undefined (so that the caller doesn't need to check undefined), so that's why I'm doing this here.